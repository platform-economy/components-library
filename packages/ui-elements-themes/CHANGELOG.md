# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
