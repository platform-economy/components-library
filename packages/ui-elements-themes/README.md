# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
