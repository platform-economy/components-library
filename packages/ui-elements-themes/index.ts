export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
