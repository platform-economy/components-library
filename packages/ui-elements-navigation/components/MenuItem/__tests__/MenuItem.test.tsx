import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import 'jest-styled-components';

import { MenuItem } from '../index';

import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-navigation', () => {
  describe('Components', () => {
    describe('MenuItem', () => {
      it('Renders MenuItem', () => {
        const menuItemClick = () => null;
        const tree = TestRenderer
          .create(withTestTheme(
            <MenuItem
              onClick={menuItemClick}
              icon={<i />}
            >
              Label
            </MenuItem>,
          ))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Renders with appropriate colors if active prop is set', () => {
        const menuItemClick = () => null;
        const tree = TestRenderer
          .create(withTestTheme(
            <MenuItem
              onClick={menuItemClick}
              icon={<i />}
              active={true}
            />,
          ))
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('Handles onClick event', () => {
        const handler = jest.fn();
        const wrapper = enzyme.mount(withTestTheme(<MenuItem icon={<i />} onClick={handler} />));

        wrapper.find('button').simulate('click');
        expect(handler).toHaveBeenCalledTimes(1);
      });

      it('Handles onClick event correctly without handler', () => {
        const wrapper = enzyme.mount(withTestTheme(<MenuItem icon={<i />} />));
        let error;

        try {
          wrapper.find('button').simulate('click');
        } catch (err) {
          error = err;
        }

        expect(error).toBe(undefined);
      });
    });
  });
});
