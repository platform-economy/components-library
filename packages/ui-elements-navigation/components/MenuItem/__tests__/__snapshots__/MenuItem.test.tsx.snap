// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-navigation Components MenuItem Renders MenuItem 1`] = `
.c0 {
  font: inherit;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50px;
  padding: 0 7px;
  margin-bottom: 2px;
  color: #text_inverted;
  background-color: #surface;
  border-width: 0 0 0 4px;
  border-style: solid;
  border-color: transparent;
  outline: none;
  cursor: pointer;
}

.c0:hover {
  color: #text;
}

.c2 {
  font-size: .875em;
  font-weight: bold;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  max-width: 128px;
  padding: 0 7px;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c1.c1.c1 {
  font-size: 1.25em;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 25px;
}

<button
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <i />
  </div>
  <div
    className="c2"
  >
    Label
  </div>
</button>
`;

exports[`ui-elements-navigation Components MenuItem Renders with appropriate colors if active prop is set 1`] = `
.c0 {
  font: inherit;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 50px;
  padding: 0 7px;
  margin-bottom: 2px;
  color: #text;
  background-color: #surface;
  border-width: 0 0 0 4px;
  border-style: solid;
  border-color: #active;
  outline: none;
  cursor: pointer;
}

.c0:hover {
  color: #text;
}

.c2 {
  font-size: .875em;
  font-weight: bold;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  max-width: 128px;
  padding: 0 7px;
  overflow: hidden;
  text-overflow: ellipsis;
}

.c1.c1.c1 {
  font-size: 1.25em;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 25px;
}

<button
  className="c0"
  onClick={[Function]}
>
  <div
    className="c1"
  >
    <i />
  </div>
  <div
    className="c2"
  />
</button>
`;
